### PR TITLE
feat: track quick-start events

### PR DESCRIPTION
**Pull request by sunnyguduru**
Added three new types of events:
`SendSetupStartedEvent`, `SendSetupSDKSelectedEvent`, `SendSetupFlagToggledEvent`

### CODE REVIEW

File: internal/quickstart/show_sdk_instructions.go [hunk 1]
```diff
@@ -92,6 +96,15 @@ func NewShowSDKInstructionsModel(
 // fetch SDK instructions
 // fetch the environment to get values to interpolate into the instructions
 func (m showSDKInstructionsModel) Init() tea.Cmd {
+	m.analyticsTracker.SendEvent(
+		m.accessToken,
+		m.baseUri,
+		"CLI Setup SDK Selected",
+		map[string]interface{}{
+			"sdk": m.canonicalName,
```
Comment (sunnyguduru) [inline internal/quickstart/show_sdk_instructions.go:104]: canonicalName seems like the preferable one here as opposed to displayName.

**Review by dbolson — CHANGES_REQUESTED** [review]
I think we can use messages and do all the tracking in the container model. That way we don't need to pass the analytics tracker into the nested models, and we can keep the tracking calls closer together.

For example,
* when we start the tea program in `quickstart.go`, it calls the `createFlagModel#Init()`
* `createFlagModel#Init()` sends a `createFlagStartedMsg` with `time.Now()`
* container receives the message in `Update`
* container calls `SendSetupStartedEvent()` with its current step (1)
* when subsequent nested models are initialized, the send their own `{model}StartedMsg`
* the container receives those messages and calls `SendSetupStartedEvent()` as needed

It would be helpful to write up a plan for the message flow before changing the code to prevent more back-and-forth.

File: internal/quickstart/choose_sdk.go [hunk 2]
```diff
@@ -64,6 +73,7 @@ func NewChooseSDKModel(selectedIndex int) tea.Model {
 // Init sends commands when the model is created that will:
 // * select an SDK if it's already been selected
 func (m chooseSDKModel) Init() tea.Cmd {
+	m.analyticsTracker.SendSetupStartedEvent(m.accessToken, m.baseUri, viper.GetBool(cliflags.OutputFlag), "2 - sdk selection")
```
[the remark this answers is not in the document]
Comment (sunnyguduru) [reply internal/quickstart/choose_sdk.go:76]: Dang, I made this mistake before. Thanks for catching it

**Comment by sunnyguduru** [comment]
> I think we can use messages and do all the tracking in the container model. That way we don't need to pass the analytics tracker into the nested models, and we can keep the tracking calls closer together.
> 
> For example,
> 
> * when we start the tea program in `quickstart.go`, it calls the `createFlagModel#Init()`
> * `createFlagModel#Init()` sends a `createFlagStartedMsg` with `time.Now()`
> * container receives the message in `Update`
> * container calls `SendSetupStartedEvent()` with its current step (1)
> * when subsequent nested models are initialized, the send their own `{model}StartedMsg`
> * the container receives those messages and calls `SendSetupStartedEvent()` as needed
> 
> It would be helpful to write up a plan for the message flow before changing the code to prevent more back-and-forth.

### SendSetupStartedEvent

- when we start the tea program in `quickstart.go`, it calls the `createFlagModel#Init()`
- `createFlagModel#Init()` sends a `createFlagStartedMsg` with `time.Now()`
- container receives the message in `Update`
- container calls `SendSetupStartedEvent()` with its current step (1)
- when subsequent nested models are initialized, the send their own `{model}StartedMsg`
- the container receives those messages and calls `SendSetupStartedEvent()` as needed

### SendSetupSDKSelectedEvent

- the container receives `choseSDKMsg` with chosen SDK of type `string`. Then container calls `SendSetupSDKSelectedEvent` with the chosen SDK.

### SendSetupFlagToggledEvent

- the container receives `toggledFlagMsg` with endTime of type `time.Time` and count of type `int`. Then container calls `SendSetupFlagToggledEvent`.

**Comment by dbolson** [comment]
Thinking about it more, the container shouldn't directly call the tracker but should send a command that does it, similar to the `toggleFlag` command. Something like
```
type eventTrackedMsg struct{}

func trackSetupStartedEvent(tracker analytics.Tracker, accessToken, baseURI string, optOut bool) tea.Cmd {
	return func() tea.Msg {
		tracker.SendWhateverEvent(accessToken, baseURI, optOut, time.Now())

		return eventTrackedMsg{}
	}
}
```
The container could use `tea.Batch()` or `tea.Sequence()` when setting `cmd` to get the current one plus the one to track the event.

**Comment by sunnyguduru** [comment]
Closing to break into smaller PRS #215

**Pull request closed without merging**